File: documentation/user/en/solve/examples/filtering-products-in-category/faceted-search-brand.cs
EvitaResponse<ISealedEntity> entities = evita.QueryCatalog(
	"evita",
	session => session.QuerySealedEntity(
        Query(
        	Collection("Product"),
        	FilterBy(
        		EntityLocaleEquals(CultureInfo.GetCultureInfo("en")),
        		HierarchyWithin(
        			"categories",
        			AttributeEquals("url", "/en/smartwatches")
        		),
        		UserFilter(
        			FacetHaving(
        				"brand",
        				EntityPrimaryKeyInSet(66465)
        			)
        		)
        	),
        	Require(
        		ReferenceSummaryOfReference(
        			"brand",
        			Impact,
        			OrderBy(
        				AttributeNatural("name", Asc)
        			),
        			EntityFetch(
        				AttributeContent("name")
        			)
        		)
        	)
        )
	)
);

File: documentation/user/en/solve/examples/filtering-products-in-category/faceted-search-brand.java
final EvitaResponse<SealedEntity> entities = evita.queryCatalog(
	"evita",
	session -> {
		return session.querySealedEntity(
			query(
				collection("Product"),
				filterBy(
					entityLocaleEquals(Locale.forLanguageTag("en")),
					hierarchyWithin(
						"categories",
						attributeEquals("url", "/en/smartwatches")
					),
					userFilter(
						facetHaving(
							"brand",
							entityPrimaryKeyInSet(66465)
						)
					)
				),
				require(
					facetSummaryOfReference(
						"brand",
						IMPACT,
						orderBy(
							attributeNatural("name", ASC)
						),
						entityFetch(
							attributeContent("name")
						)
					)
				)
			)
		);
	}
);
